src/uiveri5.js

```diff
@@ -322,19 +328,22 @@ function run(config) {
         iTimeout = iTimeout || browser.getPageTimeout;
         return browser.driver.get(sUrl, iTimeout).then(function () {
           return browser.loadUI5Dependencies();
-        }).then(function () {
-          return browser.waitForAngular();
         });
       };
 
-      browser.setLocation = function (sHash) {
-        return browser.executeAsyncScriptHandleErrors('setLocation', {
-          hash: sHash
-        }).then(function () {
-          return browser.loadUI5Dependencies();
-        }).then(function () {
-          return browser.waitForAngular();
+      browser.getProtected = function (sUrl, vAuth) {
+        vAuth = vAuth || 'auth';
+        // auth can be object with config
+        var authenticator =  moduleLoader.loadNamedModule(vAuth, [statisticCollector]);
+        // open page and login
+        browser.controlFlow().execute(function () {
+          logger.info('Opening: ' + sUrl);
         });
+        authenticator.get(sUrl);
+
+        // load waitForUI5 logic on client and
+        // ensure app is fully loaded before starting the interactions
+        return browser.loadUI5Dependencies();
```

Comment:
where is the overloading of browser.refresh() and browser.reload(). I see the tests but no impl so wondering how this is working ?

Reply:
browser.refresh: gets the current url and calls browser.get with this url

Reply:
ok but please mention in docs so the users know we support them